portindex2json/portindex2json.tcl

```diff
@@ -64,7 +64,58 @@ proc remove_extra_braces {text} {
     return [regsub -all {[\{\}]} $text ""]
 }
 
-set fd [open [lindex $argv 0] r]
+proc print_usage {} {
+    puts stdout "\nUsage: /path/to/tclsh portindex2json.tcl <path-to-portindex> \[-info key=vlaue\]"
+    puts stdout "\nEach '-info key=value' combinaton adds one key-value pair to the JSON output inside the parent key: \"info\""
+    puts stdout "\nYou may provide any number of '-info key=value' combinations."
+    puts stdout "\nThe arguments <path-to-portindex> and `-info key=value` are position independent.\n"
+}
+
+set filecount 0
+
+for {set i 0} {$i < $argc} {incr i} {
+    set arg [lindex $argv $i]
+    switch -regex -- $arg {
+        {^-.+} {
+            if {$arg eq "-info"} {
+                incr i
+                set argument [split [lindex $argv $i] =]
+                if {[lindex $argument 1] eq ""} {
+                    puts stderr "\nError: No value provided for the key: [lindex $argument 0]. The correct syntaxt is -info key=value. Use -help for more\n"
+                    exit 1
+                } else {
+                    set json_info([lindex $argument 0]) [::json::write string [lindex $argument 1]]
+                }
+            } elseif {$arg eq "-help" || $arg eq "-h" || $arg eq "--help" } {
+                print_usage
+                exit 0
+            } else {
+                puts stderr "\nError: $arg is an invalid argument. Expecting -info, -help.\n"
+                exit 1
+            }
+        }
+        default {
+            incr filecount
+            set portindex_file [lindex $argv $i]
```

Comment:
What I would do here is:
* If file was already defined, print error message, maybe usage. (No need for counter.)
* If it wasn't defined, try to catch-open it and exit with error if invalid (the code from below).

Reply:
> What I would do here is:
> 
> * If file was already defined, print error message, maybe usage. (No need for counter.)
> * If it wasn't defined, try to catch-open it and exit with error if invalid (the code from below).

Okay, thank you. I probably misinterpreted the previous suggestion and thought that before trying to catch-open any of the assignments I should check if two files were provided.

To clarify: In this case it would throw `file not found` error before it realises that two files were provided (when the first file is invalid). Is that the required functionality?

Reply:
If the first file is invalid, printing `file not found` and exiting immediately would make most sense to me.

But if the second "file" is invalid, you can probably check if `portindex_file` is already defined before trying to even read the new value, right?

Reply:
Yes, that seems fine. Thank you